### DISCUSSION TITLE
Program End but still in running state

### DISCUSSION THREAD

**Discussion by vilianch123** [arkypita/LaserGRBL#1574] [General]
Hello,arkypita!
I have run a G code files and in the end,machine send the termined characters:[MSG:Pgm End],
the responsed status is changed to IDLE,but UI status still in running state!
And I can not restart and select another G code file for maching.
Thank you!
please see attachments here:
![statue report in idle state](https://user-images.githubusercontent.com/86226681/133795239-390047ca-881d-4eec-9131-4439d67572ef.PNG)
!
![ui still in running state](https://user-images.githubusercontent.com/86226681/133795269-633bf0fc-22c8-4d6c-8625-5fb44361f602.PNG)

**Comment by arkypita** [comment]
Last gcode "M30" never receive a reply, so program is not ended by LaserGRBL point of view because we have pending commands.
Not a LaserGRBL issue but GRBL issue: https://github.com/grbl/grbl/issues/681

**Reply by vilianch123** [reply]
So the reply for "M30" is also "OK"? 
And why not use "[MSG:Pgm End]" also for program terminated? I means if "[MSG:Pgm End]" is received,program can also ended successfully.
Thank you arkypita!

**Reply by arkypita** [reply]
Yes, standard reply for each grbl command should beban 'ok' or 'error' message. In my opinion is a grbl issue.

![Screenshot_20210919_082554](https://user-images.githubusercontent.com/8782035/133917771-7eba960d-d0bc-4976-bda5-b06baebb66d0.jpg)

However I can add a better handling of M30 as well as M0 and M2 (they seem to have similar issues)

**Comment by arkypita** [comment]
What version of GRBL do you have? because with 1.1h i have no issue with M30

**Reply by arkypita** [reply]
and what controller/engraver it is?

**Reply by vilianch123** [reply]
Perhaps it is a USB line problem,I would try again.
